File: frontend/src/app/model/actividad.ts
import { Categoria } from "./categoria";
import { Proyecto } from "./proyecto";
import { Usuario } from "./usuario";

export class Actividad {

    id: String;
    proyecto: Proyecto;
    categoria: Categoria;
    fechaInicio: String;
    fechaFin: String;
    titulo: String;
    descripcion: String;
    estado: String = '1';
    usuario:Usuario;

    constructor() {
        this.proyecto = new Proyecto();
        this.categoria = new Categoria();
        this.usuario = new Usuario();
    }

}

File: backend/src/main/java/com/g4ts/agendaapp/model/Actividad.java
package com.g4ts.agendaapp.model;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "Actividad")
@Getter @Setter @ToString @Builder
@AllArgsConstructor
@NoArgsConstructor
public class Actividad {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private LocalDate fechaInicio;
    private LocalDate fechaFin;
    private String titulo;
    private String descripcion;
    private Short estado;

    @OneToOne
    @JoinColumn(name = "idProyecto")
    private Proyecto proyecto;

    @OneToOne
    @JoinColumn(name = "idCategoria")
    private Categoria categoria;

    @OneToOne
    @JoinColumn(name = "idUsuario")
    private Usuario usuario;
}
